spark3/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -367,6 +375,26 @@ private void commitWithSnapshotIsolation(OverwriteFiles overwriteFiles,
     }
   }
 
+  private class RewriteFiles extends BaseBatchWrite {
+    private final String fileSetID;
+
+    private RewriteFiles(String fileSetID) {
+      this.fileSetID = fileSetID;
+    }
+
+    @Override
+    public void commit(WriterCommitMessage[] messages) {
+      FileRewriteCoordinator coordinator = FileRewriteCoordinator.get();
+
+      Set<DataFile> newDataFiles = Sets.newHashSetWithExpectedSize(messages.length);
```

Comment:
There may be more files than the commit messages but it will be rare and it is still better to start with a rough estimate rather than with a small value and do multiple resizing steps. I think we will have at most one resizing in the current logic.